src/tasks/solvers.ts

```diff
@@ -0,0 +1,138 @@
+import "hardhat-deploy";
+import "@nomiclabs/hardhat-ethers";
+
+import { Contract, Signer } from "ethers";
+import { subtask, task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+const solversTaskList = ["add", "check", "remove"] as const;
+type SolversTasks = typeof solversTaskList[number];
+
+async function getOwnerSigner({
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment): Promise<Signer> {
+  const { owner } = await getNamedAccounts();
+  const signer = (await ethers.getSigners()).find(
+    (signer) => signer.address == owner,
+  );
+  if (signer == undefined) {
+    throw new Error(
+      "No owner found among the signers. Did you export the owner's private key?",
```

Comment:
ubernit: maybe even include did you `export PK=<your key>` or something to make it super easy to fix this error when you see it (rather than having to lookup the exact env variable name)